backend/src/auth/schemas.py

```diff
@@ -1,12 +1,108 @@
-from pydantic import BaseModel
+from pydantic import BaseModel, field_validator
 
+from src.auth.validators import email_validator, password_validator, username_validator
 
-class UserSchema(BaseModel):
+
+class UserResponse(BaseModel):
+    username: str
+    email: str
+    model_config = {
+        "json_schema_extra": {
+            "title": "User",
+            "description": "User model",
+            "example": {
+                "username": "basic",
+                "email": "basic@basic.com",
+            },
+        },
+    }
+
+
+class RegisterRequest(BaseModel):
     username: str
     email: str
     password: str
 
+    @field_validator(__field="username", mode="before")
+    @classmethod
+    def validate_username(cls, username: str) -> str:
+        return username_validator(username)
+
+    @field_validator(__field="email", mode="before")
```

Comment:
can't you just skip __field and use it positionaly? double underscore means it is private field for the function